scipy/stats/_continuous_distns.py

```diff
@@ -7166,6 +7166,92 @@ def _sf(self, x, chi):
 argus = argus_gen(name='argus', longname="An Argus Function", a=0.0, b=1.0)
 
 
+class irwinhall_gen(rv_continuous):
+    r"""
+    Irwin Hall distribution
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The `irwinhall` cumulative distribution function is:
+
+    .. math::
+
+        F(x) = \frac{1}{n!}\sum_{k=0}^{\lfloor x\rfloor}
+                    (-1)^k\binom{n}{k}(x-k)^n
+
+    for :math:`x \in [0,n]`, where :math:`n > 0` is an integer.
+    For more on the Irwin Hall distribution see [1]_.
+
+    For large :math:`n` values the calculations may be slow as most methods
+    scale at :math:`O(n)`.
+
+    References
+    ----------
+
+    .. [1] "Irwin-Hall distribution",
+           https://en.wikipedia.org/wiki/Irwin-Hall_distribution
+
+    %(after_notes)s
+
+    %(example)s
+    """
+
+    def _argcheck(self, n):
+        return (n > 0) & (np.floor(n) == n)
+
+    def _pdf_helper(x,n):
+        if x <= 0.0 or x >= n:
+            pdf_x = 0.0
+        else:
+            fl_x = int(np.floor(x)) + 1
+            kernel_list = [(-1)**k * sc.binom(n,k) * (x-k)**n
+                    for k in range(0, fl_x)]
```

Comment:
Pretty sure this can be written with numpy operations, withou list comprehensions.

Reply:
Without too much code golfing

```python
def _pdf_helper(x, n):  # PEP8 space after x,
    # Quick exit
    if not (0 < x < n):
        return 0.

    kernel_list = (np.arange(fl_x) - x)**n * sc.binom(n, np.arange(fl_x))
    kernel_list[1::2] *= -1
    return kernel_list.sum() / sc.factorial(n - 1)
```

Reply:
@ilayn if I base this off #9900 and `self.a=0.0` and `self.b=n` does it still make sense to have the quick exit here? I have no opinion either way, both should return `0.` but I'm not sure how long it takes to resolve that `x < self.a` implies `return 0.0` (say).